src/client/resources/conversions.ts

```diff
@@ -440,6 +440,11 @@ export function updateMigPlanFromValues(
   if (planValues.planClosed) {
     updatedSpec.closed = true;
   }
+  if (currentPlan.status) {
+    // if currentPlan has status, this is not 1st MigPlan reconcile.
+    // we should refresh corresponding MigStorage and MigClusters.
+    updatedSpec.refresh = true;
```

Comment:
Well done. I can't find anything wrong with this approach. This makes the UI code much more readable.  LGTM.